src/Bicep.Core/Analyzers/Linter/Rules/SecureParamsInNestedDeploymentsRule.cs

```diff
@@ -0,0 +1,141 @@
+                        }
+                    }
+                }
+            }
+        }
+
+        private static ObjectSyntax? GetPropertieIfOuterScopedDeployment(ResourceSymbol resource)
+        {
+            if (resource.TryGetResourceType() is ResourceType resourceType)
+            {
+                if (resourceType.TypeReference.FormatType().Equals("microsoft.resources/deployments", LanguageConstants.ResourceTypeComparison))
+                {
+                    if (resource.TryGetBodyProperty(LanguageConstants.ResourcePropertiesPropertyName)?.Value is ObjectSyntax propertiesObject)
+                    {
+                        // It's a valid deployment resource
+                        if (propertiesObject is not null && propertiesObject.TryGetPropertyByNameRecursive(new[] { "expressionEvaluationOptions", "scope" })
+                            ?.Value
+                            is StringSyntax scope)
+                        {
+                            if (scope.TryGetLiteralValue()?.Equals("inner", StringComparison.InvariantCultureIgnoreCase) == true)
+                            {
+                                // Scope is explicitly marked as "inner"
+                                return null;
+                            }
+                        }
+
+                        // Default scope is outer if not explicitly marked as inner
+                        return propertiesObject;
+                    }
+                    else
+                    {
+                        // No properties object (not valid)
+                        return null;
+                    }
+                }
+                else
+                {
+                    // Not a deployment resource
+                    return null;
+                }
+            }
+            else
+            {
+                return null;
+            }
+        }
+        private static Symbol[] FindSecureParametersReferencedInNestedDeployment(SemanticModel model, ResourceSymbol deploymentResource)
+        {
+            var secureParamsReferencedInDeployment = model.Root.ParameterDeclarations
+                .Where(p => p.NameSyntax.IsValid && p.IsSecure())
+                .Where(p => model.FindReferences(p, deploymentResource.DeclaringSyntax).Any());
+            return secureParamsReferencedInDeployment.ToArray();
+        }
+
+        private static IEnumerable<SyntaxBase> FindListFunctionReferencesInSyntaxTree(SemanticModel model, SyntaxBase syntaxTree)
+        {
+            return SyntaxAggregator.Aggregate(syntaxTree, current =>
+            {
+                if (model.GetSymbolInfo(current) is FunctionSymbol symbol)
+                {
+                    if (SemanticModelHelper.TryGetFunctionInNamespace(model, AzNamespaceType.BuiltInName, current) is FunctionCallSyntaxBase functionCall
+                        && functionCall.Name.IdentifierName.StartsWith("list"))
```

Comment:
Can we moce ListFunctionPrefix in FindPossibleSecretsVisitor to LanguageConstants and change this to below?
```
functionCall.Name.IdentifierName.StartsWithOrdinalInsensitively(ListFunctionPrefix)
```

Reply:
good idea, thanks